### ISSUE TITLE
Race condition in tag selector tests

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#1659]
I'm getting intermittent failures in tag selector tests, generally from `getRegularTags()` returning incorrect results. E.g.:

```
    ✖ [FAIL] should add a tag when an item is added in a collection
      AssertionError: expected 5 to equal 0 at
        @resource://zotero-unit-tests/tagSelectorTest.js:233:4
```

Failing tests work individually, but I get failures if I run `./test/runtests.sh -b -f tagSelector`, so I suspect `waitForTagSelector()` isn't properly waiting for tag selector refreshes after Reactification.

(I can reproduce this in 3c6ab77e44a9a4fe4858643d07f8f147929126fd, before my recent changes.)

**Comment by adomasven** [comment]
I am not seeing any race conditions, but I've made some changes that hopefully will make this more robust

**Comment by dstillman** [comment]
I'm still getting a race condition. On first run, it failed at the same line. Second time worked. Third time failed again. With debug output, which slows it down, it succeeds.

We can add a delay to `waitForTagSelector()` if we have to, but it'd be nice if there were a more reliable solution, since it's likely to come up for tests in other React components, and we shouldn't have to add in multiple `waitForTagSelector()` calls to account for the internals of the notification handler. (It's also not specific to this test, so we should revert those extra calls — a delay would be a better fix than that.) Do you have any idea why this might be happening? Does this have something to do with batched `setState()` calls? Something else?

**Comment by adomasven** [comment]
Certain operations issue multiple notifications which then cause multiple `setState()` calls (which `waitForTagSelector()` waits for).

**Comment by dstillman** [comment]
This seems to be fixed for me after b20bf34. We can reopen if it appears anywhere else.

**Comment by dstillman** [comment]
(Though the fact that e8f1864 doesn't cause a hang means that it's updating multiple times, which isn't ideal. As I understand it, React batches `setState()` automatically in some cases, but maybe only when triggered by events. If it's not doing that in cases like this, we may want to debounce the `setState()` calls ourselves in some situations, since it's not uncommon for multiple Notifier events to be triggered at the same time.)
